openmls/src/treesync/node/leaf_node.rs

```diff
@@ -545,123 +571,200 @@ impl LeafNode {
         }
     }
 
-    /// Returns `true` if the [`ExtensionType`] is supported by this leaf node.
-    pub(crate) fn supports_extension(&self, extension_type: &ExtensionType) -> bool {
-        self.payload
-            .capabilities
-            .extensions
-            .iter()
-            .any(|et| et == extension_type)
-            || default_extensions().iter().any(|et| et == extension_type)
+    /// Returns the [`Lifetime`] if present.
+    /// `None` otherwise.
+    pub(crate) fn life_time(&self) -> Option<&Lifetime> {
+        if let LeafNodeSource::KeyPackage(life_time) = &self.payload.leaf_node_source {
+            Some(life_time)
+        } else {
+            None
+        }
     }
 
-    /// Returns `true` if the [`ProposalType`] is supported by this leaf node.
-    pub(crate) fn supports_proposal(&self, proposal_type: &ProposalType) -> bool {
-        self.payload
-            .capabilities
-            .proposals
-            .iter()
-            .any(|pt| pt == proposal_type)
-            || default_proposals().iter().any(|pt| pt == proposal_type)
+    /// Returns a reference to the [`Signature`] of this leaf.
+    pub fn signature(&self) -> &Signature {
+        &self.signature
     }
 
-    /// Check that all extensions that are required, are supported by this leaf
-    /// node.
+    /// Return a reference to [`Capabilities`].
+    pub(crate) fn capabilities(&self) -> &Capabilities {
+        &self.payload.capabilities
+    }
+
+    /// Return a mutable reference to [`Capabilities`].
+    #[cfg(test)]
+    pub fn capabilities_mut(&mut self) -> &mut Capabilities {
+        &mut self.payload.capabilities
+    }
+
+    /// Return a reference to the leaf node extensions.
+    pub fn extensions(&self) -> &Extensions {
+        &self.payload.extensions
+    }
+
+    // ----- Validation ----------------------------------------------------------------------------
+
+    /// Check that lifetime is acceptable.
+    ///
+    /// * Check that lifetime range is acceptable.
+    /// * Check that current time is between `Lifetime.not_before` and `Lifetime.not_after`.
+    pub(crate) fn validate_lifetime(&self) -> Result<&Self, LeafNodeValidationError> {
+        // self.validate_lifetime_range()?; // TODO(openmls/annotations#36)
+        self.validate_lifetime_actuality()?;
+
+        Ok(self)
+    }
+
+    /// Check that lifetime range is acceptable.
+    #[allow(unused)] // TODO(openmls/annotations#36)
```

Comment:
Why are we not using this in the function above? I'm sure there is a good reason, so it would be great if you could briefly explain it either here or in openmls/annotations#36.

Reply:
Resolved by rebase and adding of `TODO`.